merlin/models/tf/blocks/core/transformations.py

```diff
@@ -37,6 +37,12 @@ def call(self, inputs: TabularData, **kwargs) -> TabularData:
             if isinstance(val, tuple):
                 values = val[0][:, 0]
                 row_lengths = val[1][:, 0]
+
+                if values.dtype == tf.float32:
```

Comment:
Are we always converting values from float to int? Couldn't the sparse features be floats?